org-mode/lib/Data/Org.hs

```diff
@@ -125,7 +143,64 @@ data Block
   | Table (NonEmpty Row)
   | Paragraph (NonEmpty Words)
   deriving stock (Eq, Show, Generic)
-  deriving anyclass (Hashable)
+
+-- | An org-mode timestamp. Must contain at least a year-month-day and the day
+-- of the week:
+--
+-- @
+-- \<2021-04-27 Tue\>
+-- @
+--
+-- but also may contain a time:
+--
+-- @
+-- \<2021-04-27 Tue 12:00\>
+-- @
+--
+-- or a time range:
+--
+-- @
+-- \<2021-04-27 Tue 12:00-13:00\>
+-- @
+--
+-- and/or a repeater value:
+--
+-- @
+-- \<2021-04-27 Tue +1w\>
+-- @
+data OrgDateTime = OrgDateTime
+  { dateDay       :: Day
+  , dateDayOfWeek :: DayOfWeek
+  , dateTime      :: Maybe OrgTime
+  , dateRepeat    :: Maybe Repeater }
+  deriving stock (Eq, Show)
+
+-- | The time portion of the full timestamp. May be a range, as seen in the
+-- following full timestamp:
+--
+-- @
+-- \<2021-04-27 Tue 12:00-13:00\>
+-- @
+data OrgTime = OrgTime
+  { timeStart :: TimeOfDay
+  , timeEnd   :: Maybe TimeOfDay }
+  deriving stock (Eq, Show)
+
+-- | An indication of how often a timestamp should be automatically reapplied in
+-- the Org Agenda.
+data Repeater = Repeater
+  { repDirection :: TimeDirection
+  , repValue     :: Word
```

Comment:
What's the semantics of `repValue`?

Reply:
A repeater is something like `+2w`. In this case, the `repValue` is 2.